tsconfig.json

```diff
@@ -4,6 +4,7 @@
  */
 {
   "compilerOptions": {
+    "target": "ES6",
```

Comment:
this is recommended by the oracle

Reply:
why? why not nodenext? does this affect our emitted declarations at all?

Reply:
for context (from typescript [docs](https://www.typescriptlang.org/tsconfig/#target)):

```
Modern browsers support all ES6 features, so ES6 is a good choice. You might choose to set a lower target if your code is deployed to older environments, or a higher target if your code is guaranteed to run in newer environments.

The target setting changes which JS features are downleveled and which are left intact. For example, an arrow function () => this will be turned into an equivalent function expression if target is ES5 or lower.
```

However, since we are not generating any JavaScript (only type declarations), this doesn't affect the output of `tsc`, but it might influence the type system and available built-in types, so we should add it.